08-js-security/level2/index.secure.html

```diff
@@ -0,0 +1,91 @@
+ecu<!doctype html>
+<html>
+  <head>
+    <!-- Internal game scripts/styles, mostly boring stuff -->
+    <script src="https://xss-game.appspot.com/static/game-frame.js"></script>
+    <link rel="stylesheet" href="https://xss-game.appspot.com/static/game-frame-styles.css" />
+ 
+    <!-- This is our database of messages -->
+    <script src="https://xss-game.appspot.com/static/post-store.js"></script>
+   
+    <script>
+      var defaultMessage = "Welcome!<br><br>This is your <i>personal</i>"
+        + " stream. You can post anything you want here, especially "
+        + "<span style='color: #f00ba7'>madness</span>.";
+ 
+      var DB = new PostDB(defaultMessage);
+ 
+      function displayPosts() {
+        var containerEl = document.getElementById("post-container");
+        containerEl.innerHTML = "";
+ 
+        var posts = DB.getPosts();
+        for (var i=0; i<posts.length; i++) {
+          var html = '<table class="message"> <tr> <td valign=top> '
+            + '<img src="https://xss-game.appspot.com/static/level2_icon.png"> </td> <td valign=top '
+            + ' class="message-container"> <div class="shim"></div>';
+ 
+          html += '<b>You</b>';
+          html += '<span class="date">' + new Date(posts[i].date) + '</span>';
+          html += "<blockquote>" + posts[i].message + "</blockquote";
```

Comment:
My guess is server-to-client fix should appear somewhere in this line :)
BTW, do not leave the closing tags unclosed, especially if it's a blockquote!